sklearn/tree/_criterion.pyx

```diff
@@ -596,6 +602,12 @@ cdef class Gini(ClassificationCriterion):
               = 1 - \sum_{k=0}^{K-1} count_k ** 2
     """
 
+    def __reduce__(self):
```

Comment:
Why did we not need these before? Is it not enough to use `type(self)` as the first member of the tuple in `{Classification,Regression}Criterion`?